integration_tests/src/main/python/cast_test.py

```diff
@@ -167,3 +166,8 @@ def test_cast_long_to_decimal_overflow():
             f.col('a').cast(DecimalType(18, -1))),
         conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
 
+@pytest.mark.skip(not is_before_spark_311(), reason="Only in Spark 3.1.1+ do we have incompatibility with negative decimal scale")
```

Comment:
Is this an incompatibility or simply that negative scale decimals aren't supported in Spark 3.1.1+?

Reply:
There was a bug introduced in Spark 3.1.1 that has lead to us being incompatible with Spark. I have filed an issue [here](https://issues.apache.org/jira/browse/SPARK-37451). Let me know if this still isn't clear by this message